engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -235,16 +237,18 @@ private void checkWorkingDirectory(Optional<TruffleFile> maybeProjectRoot) {
       var cwd = environment.getCurrentWorkingDirectory().getAbsoluteFile().normalize();
       try {
         if (!cwd.isSameFile(parent)) {
+          var maskedCwd = MaskedPath$.MODULE$.apply(Path.of(cwd.toString()));
           var maskedPath = MaskedPath$.MODULE$.apply(Path.of(parent.toString()));
-          logger.log(
-              Level.WARNING,
-              "Initializing the context in a different working directory than the one containing"
-                  + " the project root. This may lead to relative paths not behaving as advertised"
-                  + " by `File.new`. Please run the engine inside of `{0}` directory.",
-              maskedPath);
+          var templ =
+              "Initializing with unexpected working directory (%s). "
+                  + "This may lead to improper relative paths resolution by `File.new`. "
+                  + "Change working directory to %s and the engine again.";
+          var msg = templ.formatted(maskedCwd, maskedPath);
+          logger.log(Level.WARNING, msg);
+          assert false : msg;
```

Comment:
Forcing this is OK. But add a description to the PR how to bring back the old behavior for our fellow developers.

Reply:
There is:

> by freaking out on wrong current working directory in development mode
>
>    only then the message will get some attention
>    and inspire developers to address it
>
> however there are cases where we do not care about the CWD being correct
>
>    let's add CHECK_CWD option
>    and disable it in JUnit tests & co.

What else would you like to see?

Reply:
`env JAVA_TOOL_OPTIONS='-Dpolyglot.enso.checkCurrentDirectory=false'`